### PR TITLE
refactor(divider): refactor with ts

### PR DESCRIPTION
**Pull request by John60676**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/divider/index.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+const Divider = defineComponent({
+  name: 'ADivider',
+  props: {
+    prefixCls: String,
+    type: {
+      type: String as PropType<'horizontal' | 'vertical' | ''>,
+      default: 'horizontal',
+    },
+    dashed: Boolean,
```
Comment (xrkffgg) [inline components/divider/index.tsx:12]: 这个应该也有一个默认值

Comment (John60676) [reply components/divider/index.tsx]: done

**Comment by xrkffgg** [comment]
有冲突，rebase 下，push -f

File: components/divider/index.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      default: false,
+    },
+    orientation: {
+      type: String as PropType<'left' | 'right' | 'center'>,
+      default: 'center',
+    },
+  },
+  setup(props, { slots }) {
+    const configProvider = inject('configProvider', ConfigConsumerProps);
+    const getPrefixCls = configProvider.getPrefixCls;
```
Comment (Amour1688) [inline components/divider/index.tsx:23]: 用解构吧

File: components/divider/index.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        [`${prefixClsRef}-${type}`]: true,
+        [`${prefixClsRef}-with-text${orientationPrefix}`]: slots.default,
+        [`${prefixClsRef}-dashed`]: !!dashed,
+      };
+    });
+
+    return () => {
+      return (
+        <div class={classString.value} role="separator">
+          {<span class={`${prefixCls.value}-inner-text`}>{slots.default?.()}</span>}
```
Comment (Amour1688) [inline components/divider/index.tsx:41]: `{` 可以去掉

File: components/divider/index.tsx (hunk 2 above)
Comment (John60676) [reply components/divider/index.tsx]: done

File: components/divider/index.tsx (hunk 3 above)
Comment (John60676) [reply components/divider/index.tsx]: done

**Pull request merged**